api/releases.go

```diff
@@ -80,8 +87,7 @@ func SearchReleasesHandlerFunc(validator QueryParamValidator, builder ReleaseQue
 				SortBy:         sort.(query.Sort),
 				ReleasedAfter:  fromDate.(query.Date),
 				ReleasedBefore: toDate.(query.Date),
-				Upcoming:       upcoming,
-				Published:      !upcoming,
+				Type:           relType.(query.ReleaseType),
```

Comment:
(minor): I think we need to do something like 
```
if rt, ok := relType.(query.ReleaseType); !ok{
   return
}
```

Reply:
this shouldn't be necessary - at this point relType is validated and is guaranteed to be a query.ReleaseType